File: common/component/protcol/c#/caller/gate_call_hubcaller.cs
/*this caller file is codegen by juggle for c#*/
using System;
using System.Collections;
using System.IO;

namespace caller
{
    public class gate_call_hub : juggle.Icaller 
    {
        public gate_call_hub(juggle.Ichannel _ch) : base(_ch)
        {
            module_name = "gate_call_hub";
        }

        public void reg_hub_sucess()
        {
            ArrayList _argv = new ArrayList();
            call_module_method("reg_hub_sucess", _argv);
        }

        public void client_call_hub(String argv0,String argv1,String argv2,ArrayList argv3)
        {
            ArrayList _argv = new ArrayList();
            _argv.Add(argv0);
            _argv.Add(argv1);
            _argv.Add(argv2);
            _argv.Add(argv3);
            call_module_method("client_call_hub", _argv);
        }

    }
}

File: common/protcol/js/caller/gate_call_hubcaller.js
/*this caller file is codegen by juggle for js*/
function gate_call_hub_caller(ch){
    Icaller.call(this, "gate_call_hub", ch);

    this.reg_hub_sucess = function(){
        var _argv = [];
        this.call_module_method.call(this, "reg_hub_sucess", _argv);
    }

    this.client_connect = function( argv0){
        var _argv = [argv0];
        this.call_module_method.call(this, "client_connect", _argv);
    }

    this.client_disconnect = function( argv0){
        var _argv = [argv0];
        this.call_module_method.call(this, "client_disconnect", _argv);
    }

    this.client_exception = function( argv0){
        var _argv = [argv0];
        this.call_module_method.call(this, "client_exception", _argv);
    }

    this.client_call_hub = function( argv0, argv1, argv2, argv3){
        var _argv = [argv0,argv1,argv2,argv3];
        this.call_module_method.call(this, "client_call_hub", _argv);
    }

}
(function(){
    var Super = function(){};
    Super.prototype = Icaller.prototype;
    gate_call_hub_caller.prototype = new Super();
})();
gate_call_hub_caller.prototype.constructor = gate_call_hub_caller;

